test/utility.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+  SLB.Utility
+
+  Copyright Michael Park, 2017
+  Copyright Agustin Berge, 2017
+
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+*/
+
+#include <slb/utility.hpp>
+
+#include <cstddef>
+#include <utility>
+
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#if __cpp_lib_integer_sequence ||                                              \
+    (_LIBCPP_VERSION >= 1101 && _LIBCPP_STD_VER > 11) || defined(_MSC_VER)
+#define HAS_STD_INTEGER_SEQUENCE 1
+#else
+#define HAS_STD_INTEGER_SEQUENCE 0
+#endif
+
+#if HAS_STD_INTEGER_SEQUENCE
+template <typename T, T... Is>
+std::size_t deduce_std_integer_sequence(std::integer_sequence<T, Is...>) {
+  return sizeof...(Is);
+}
+#endif
+
+TEST_CASE("index_sequence", "[utility.syn]") {
+  CHECK(std::is_same<slb::index_sequence<>,
+                     slb::integer_sequence<std::size_t>>::value);
+
+  CHECK(std::is_same<slb::index_sequence<0>,
+                     slb::integer_sequence<std::size_t, 0>>::value);
+
+  CHECK(std::is_same<slb::index_sequence<1, 0>,
+                     slb::integer_sequence<std::size_t, 1, 0>>::value);
+
+  CHECK(std::is_same<slb::index_sequence<2, 1, 0>,
+                     slb::integer_sequence<std::size_t, 2, 1, 0>>::value);
+
+#if HAS_STD_INTEGER_SEQUENCE
+  /* std-compatible */ {
+    slb::index_sequence<101, 202, 303, 404> slb_is;
+    std::index_sequence<101, 202, 303, 404> std_is = slb_is;
+    (void)std_is;
+
+    CHECK(deduce_std_integer_sequence(slb_is) == 4);
+  }
+#endif
+}
+
+TEST_CASE("make_index_sequence", "[utility.syn]") {
+  CHECK(
+      std::is_same<slb::make_index_sequence<0>, slb::index_sequence<>>::value);
+
+  CHECK(
+      std::is_same<slb::make_index_sequence<1>, slb::index_sequence<0>>::value);
+
+  CHECK(std::is_same<slb::make_index_sequence<2>,
+                     slb::index_sequence<0, 1>>::value);
+
+  CHECK(std::is_same<slb::make_index_sequence<3>,
+                     slb::index_sequence<0, 1, 2>>::value);
+
+#if HAS_STD_INTEGER_SEQUENCE
+  /* std-compatible */ {
+    slb::make_index_sequence<4> slb_is;
+    std::make_index_sequence<4> std_is = slb_is;
+    (void)std_is;
+
+    CHECK(deduce_std_integer_sequence(slb_is) == 4);
+  }
```

Comment:
These compatibility tests should be subsumed by the `index_sequence` compatibility tests, given that the tests above guarantee that the alias is indeed that type.